infra/vpcs/vpc/main.tf

```
@@ -0,0 +1,58 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+locals {
+  # Choose the region where this infrastructure should be deployed.
+  region = module.project_config.default_region
+
+  # Set project tags that will be used to tag all resources. 
+  tags = merge(module.project_config.default_tags, {
+    description = "A custom VPC"
+  })
+  # change this if you plan to have multiple vpc's
+  vpc_name = "${module.project_config.project_name}-vpc"
+}
+
+terraform {
+
+  required_version = "~>1.2.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~>4.20.1"
+    }
+  }
+
+  # Terraform does not allow interpolation here, values must be hardcoded.
+  backend "s3" {
+    bucket         = "<TF_STATE_BUCKET_NAME>"
+    dynamodb_table = "<TF_LOCKS_TABLE_NAME>"
+    key            = "infra/vpc.tfstate"
```

nit: could we mirror the module structure name ie.
```suggestion
    key            = "infra/networks/vpc.tfstate"
```